src/python/zquantum/core/wip/circuits/conversions/qiskit_conversions.py

```diff
@@ -303,7 +305,12 @@ def _import_controlled_qiskit_op(
     wrapped_qubits = qiskit_qubits[qiskit_gate.num_ctrl_qubits :]
     wrapped_op = _import_qiskit_op(qiskit_gate.base_gate, wrapped_qubits)
     qubit_indices = map(_import_qiskit_qubit, qiskit_qubits)
-    return wrapped_op.gate.controlled(qiskit_gate.num_ctrl_qubits)(*qubit_indices)
+    if isinstance(wrapped_op, _gates.GateOperation):
+        return wrapped_op.gate.controlled(qiskit_gate.num_ctrl_qubits)(*qubit_indices)
+    else:
+        raise NotImplementedError(
```

Comment:
I'm not sure about this as well.

Reply:
mypy wasn't happy with the original code because `wrapped_op` could be either be a `GateOperation` or a `AnonGateOperation`, but only `GateOperation` has a `gate` attribute. So I think the original behavior would have been to raise an `AttributeError` when `wrapped_op` is a `AnonGateOperation`, whereas now it raises `NotImplementedError`. @dexter2206 or @alexjuda what are your thoughts?

Reply:
Also note that I updated `_import_qiskit_op` to handle the `NotImplementedError`. @dexter2206 @alexjuda would be good if you could confirm whether this makes sense to do. https://github.com/zapatacomputing/z-quantum-core/blob/ce48d66e5b03ba10b0e29eb4da4466da4352e4b0/src/python/zquantum/core/wip/circuits/conversions/qiskit_conversions.py#L260-L273

Reply:
You've found a bug! 🙂 Turns out I forgot about adding (and testing) support for converting controlled custom gates between our circuits and Qiskit. I'll be happy to fix it, but I'd rather do that in another PR so this one doesn't get convoluted. 

My suggestion would be to leave the `raise NotImplementedError(...)` here + abstain from catching the error around line 270 to let it fail explicitly until I fix it. Would that make sense?

Reply:
Makes sense! Removed the catching of `NotImplementedError` in 0bf1cfdbfc7ea1e83c71799e33a4d580a46d3fe8